### DISCUSSION TITLE
Setting the focus does not work if there is an update on UI elements.

### DISCUSSION THREAD

**Discussion by hgfrey** [primefaces/community#931] [PrimeFaces] [answered]
In the spirit of accessibility, I try to focus on the message close icon when messages are displayed. I try this from the backend with this call:

`PrimeFaces.current().executeScript("$('.ui-messages-close').focus();");`

This works great as long as no update attribute is added to the button element in the xhtml page.

Any idea how I can set the focus and still update the UI?

Thank you.

**Comment by melloware** [comment, marked as answer]
Try...

```java
PrimeFaces.current().executeScript("setTimeout(() => $('.ui-messages-close').focus(), 500);");

**Reply by melloware** [reply]
@hgfrey can you report back so i can mark this as answered or not?
